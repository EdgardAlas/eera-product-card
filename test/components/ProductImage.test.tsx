import React from 'react';
import renderer from 'react-test-renderer';
import ProductCard, { ProductImage } from '../../src/components';
import { product1 } from '../data/products';

describe('ProductImage', () => {
  test('debe de mostrar el componente con la imagen personalizada', () => {
    const wrapper = renderer.create(
      <ProductImage img="https://via.placeholder.com/150" />
    );
    expect(wrapper.toJSON()).toMatchSnapshot();
  });

  test('debe de mostrar la imagendel producto', () => {
    const wrapper = renderer.create(
      <ProductCard product={product1}>{() => <ProductImage />}</ProductCard>
    );
    expect(wrapper.toJSON()).toMatchSnapshot();
  });
});
