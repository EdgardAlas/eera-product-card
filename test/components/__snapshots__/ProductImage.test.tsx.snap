// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductImage debe de mostrar el componente con la imagen personalizada 1`] = `
<img
  alt="Product"
  className="productImg "
  src="https://via.placeholder.com/150"
/>
`;

exports[`ProductImage debe de mostrar la imagendel producto 1`] = `
<div
  className="productCard "
>
  <img
    alt="Product"
    className="productImg "
    src="./coffee-mug.png"
  />
</div>
`;
